src/sentry/api/endpoints/project_key_stats.py

```diff
@@ -30,35 +38,75 @@ def get(self, request: Request, project, key_id) -> Response:
         except ProjectKey.DoesNotExist:
             raise ResourceDoesNotExist
 
+        # Outcomes queries are coupled to Django's QueryDict :(
+        query_data = QueryDict(mutable=True)
```

Comment:
Can we break this out into a `query_data_builder` helper function?

Reply:
Probably could. I would prefer to build queries using simple dictionaries instead of coupling to django's QueryDict. I could do that as a follow up to this change as this is the 3rd time I've moved tsdb queries to outcomes.